### PR TITLE
Task5/flux redux

### PR DESCRIPTION
**Pull request by OMManko**

### CODE REVIEW

File: src/actions/actions.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    type: actionTypes.SEARCH_BY,
+    option
+});
+
+export const changeSortOption = option => ({
+    type: actionTypes.SORT_BY,
+    option
+});
+
+export const getSelectedMovie = movie => ({
```
Comment (nsasinovich) [inline src/actions/actions.js]: I'm not sure you're getting a movie. You rather set it (to store I mean)

File: src/actions/actions.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export const updateInputValue = value => ({
+    type: actionTypes.UPDATE_INPUT_VALUE,
+    value
+});
+
+export const fetchMovies = (inputValue, filter) => (dispatch) => {
+    return fetch(`${endPoint}?searchBy=${filter}&search=${inputValue}`)
+        .then(res => res.json())
+        .then(data => dispatch(showResults(data)))
```
Comment (nsasinovich) [inline src/actions/actions.js]: The same thing with the naming here. You're not showing the results, you're setting them to store and components themselves decide to show them or not (like what if the results are empty?)

File: src/actions/actions.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        .catch((e) => {
+            console.log(e);
+        });
+};
+
+export const fetchMovie = (movieId) => (dispatch) => {
+    return fetch(`${endPoint}/${movieId}`)
+        .then(res => res.json())
+        .then(json => dispatch(getSelectedMovie(json)))
+        .catch((e) => {
```
Comment (nsasinovich) [inline src/actions/actions.js]: as you're not performing anything else in catch it can be simplified like `.catch(console.log)`

File: src/actions/actions.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        .then(data => dispatch(showResults(data)))
+        .catch((e) => {
+            console.log(e);
+        });
+};
+
+export const fetchMovie = (movieId) => (dispatch) => {
+    return fetch(`${endPoint}/${movieId}`)
+        .then(res => res.json())
+        .then(json => dispatch(getSelectedMovie(json)))
```
Comment (nsasinovich) [inline src/actions/actions.js]: `getSelectedMovie` think about renaming it

File: src/components/movieCard/movieCard.jsx [hunk 5]
```diff
@@ -4,28 +4,42 @@ import "./movieCard.scss";
...
+    selectMovie: PropTypes.func
 };
 
 MovieCard.defaultProps = {
     year: 2000,
-    genre: "Not determined"
+    genres: ['Not determined']
+};
+
+const mapStateToProps = state => ({
```
Comment (nsasinovich) [inline src/components/movieCard/movieCard.jsx]: `mapStateToProps` is redundant

File: src/components/movieList/movieList.jsx [hunk 6]
```diff
@@ -3,28 +3,32 @@ import { hot } from "react-hot-loader";
 import PropTypes from "prop-types";
 import MovieCard from "../movieCard/movieCard";
 import "./movieList.scss";
+import { connect } from "react-redux";
 
-const MovieList = ({ movies }) => (
+const MovieList = ({ moviesList }) => (
     <div className="movieList">
-        {movies.map((movie, i) => {
+        {moviesList.map((movie, i) => {
```
Comment (nsasinovich) [inline src/components/movieList/movieList.jsx]: please add default value for movielist

File: src/components/searchResults/searchResults.jsx [hunk 7]
```diff
@@ -3,65 +3,50 @@ import { hot } from "react-hot-loader";
...
+            selectedSortOption
+        } = this.props;
+
+
+        const formattedMoviesList = moviesList.map((movie) => {
+            movie.release = new Date(movie.release_date).getUTCFullYear();
+            return movie;
+        });
+
+        moviesList.sort((option1, option2) => option2[selectedSortOption] - option1[selectedSortOption]);
```
Comment (nsasinovich) [inline src/components/searchResults/searchResults.jsx]: shouldn't you sort `formattedMoviesList`?

Comment (nsasinovich) [reply src/components/searchResults/searchResults.jsx]: In fact, props are [read-only](https://reactjs.org/docs/components-and-props.html#props-are-read-only) so you cannot modify them

File: src/components/searchResultsAddInfoPanel/searchResultsAddInfoPanel.jsx [hunk 8]
```diff
@@ -3,46 +3,86 @@ import { hot } from "react-hot-loader";
...
+                            <div className="searchResultsAddInfo">
+                                <p>{total} movies found</p>
+                                <RadioGroup title="Sort by"
+                                    name="sortByField"
+                                    handleAction={changeCurrentSortOption}
+                                    radios={sortOptionsGroup}
+                                />
+                            </div>
+                        )}
+                    {Object.keys(selectedMovieInfo).length > 0 &&
```
Comment (nsasinovich) [inline src/components/searchResultsAddInfoPanel/searchResultsAddInfoPanel.jsx]: `Object.keys(selectedMovieInfo).length > 0` can be moved to a separate const as you use it twice

File: src/components/shared/input/input.jsx [hunk 9]
```diff
@@ -1,13 +1,37 @@
...
 import "./input.scss";
 import PropTypes from "prop-types";
+import { updateInputValue } from "../../../actions/actions";
+import { hot } from "react-hot-loader";
+import { connect } from "react-redux";
 
-const TextField = ({ className }) => (
-    <input type="text" className={className}/>
-);
+class TextField extends React.Component {
```
Comment (nsasinovich) [inline src/components/shared/input/input.jsx]: Looks like functional component, why did you change it to class?

File: src/components/shared/radioGroup/radioGroup.jsx [hunk 10]
```diff
@@ -6,7 +6,7 @@ import "./radioGroup.scss";
 
 class RadioGroup extends React.Component {
```
Comment (nsasinovich) [inline src/components/shared/radioGroup/radioGroup.jsx]: Can be functional too

File: src/components/shared/input/input.jsx (hunk 9 above)
Comment (OMManko) [reply src/components/shared/input/input.jsx]: Fixed

File: src/components/shared/radioGroup/radioGroup.jsx (hunk 10 above)
Comment (OMManko) [reply src/components/shared/radioGroup/radioGroup.jsx]: Fixed

File: src/actions/actions.js (hunk 1 above)
Comment (OMManko) [reply src/actions/actions.js]: Done

File: src/actions/actions.js (hunk 2 above)
Comment (OMManko) [reply src/actions/actions.js]: Done

File: src/actions/actions.js (hunk 4 above)
Comment (OMManko) [reply src/actions/actions.js]: Done

File: src/components/movieCard/movieCard.jsx (hunk 5 above)
Comment (OMManko) [reply src/components/movieCard/movieCard.jsx]: removed

File: src/actions/actions.js (hunk 3 above)
Comment (OMManko) [reply src/actions/actions.js]: done

File: src/components/searchResultsAddInfoPanel/searchResultsAddInfoPanel.jsx (hunk 8 above)
Comment (OMManko) [reply src/components/searchResultsAddInfoPanel/searchResultsAddInfoPanel.jsx]: done

File: src/components/searchResults/searchResults.jsx (hunk 7 above)
Comment (OMManko) [reply src/components/searchResults/searchResults.jsx]: Fixed

File: src/components/movieCard/movieCard.jsx [hunk 11]
```diff
@@ -35,11 +35,8 @@ MovieCard.defaultProps = {
...
 
-const mapStateToProps = state => ({
-});
-
 const mapDispatchToProps = {
     selectMovie: fetchMovie
 };
 
-export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(MovieCard));
+export default hot(module)(connect('', mapDispatchToProps)(MovieCard));
```
Comment (nsasinovich) [inline src/components/movieCard/movieCard.jsx]: Please use `null` for this case

Comment (OMManko) [reply src/components/movieCard/movieCard.jsx]: done
